src/python/pants/rules/core/test.py

```diff
@@ -103,13 +114,23 @@ class AddressAndDebugRequest:
     address: Address
     request: TestDebugRequest
 
+@union
+@dataclass(frozen=True)
+class CoverageDataBatch:
```

Comment:
I like the word `Batch` more than `Collection`. Good idea. Also it's less overloaded than `Collection`.

Reply:
Stu's idea really :)